### PR TITLE
Revise use cases with transformers

### PR DESCRIPTION
**Pull request by anssiko**
Add new use cases:
- Text-to-image
- Speech Recognition
- Text Generation
- (Image Segmentation now refers to [SegAny] also)

Add new reference models:
- Text-to-image: stable-diffusion-v1-5
- Image segmentation: segment-anything
- Speech-to-text: whisper-tiny.en
- Text generation: t5-small, m2m100_418M, gpt2, llama-2-7b

Remove redundant local reference: [POWERFUL-FEATURES]

***
<a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/pull/507.html" title="Last updated on Jan 18, 2024, 7:41 PM UTC (1d0f5c0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/webmachinelearning/webnn/507/50a00b9...1d0f5c0.html" title="Last updated on Jan 18, 2024, 7:41 PM UTC (1d0f5c0)">Diff</a>

### CODE REVIEW

**Comment by anssiko** [comment]
This is to revise the use cases section with transformers ahead our upcoming "delta" wide review for the new content added since our initial CR in March 2023.

@dontcallmedom please check the use cases are representative of use cases that resonate with our wide review audience. I did not restructure the old material, but plugged in the new transformers-enabled use cases into the existing list. I note it is possible to create a higher-level taxonomy for these use cases but I'll leave that as future work for now. Also check if you can find why PR Preview failed to build. Builds cleanly locally.

@zolkis please check in particular the referenced papers are the canonical ones and the latest, and that the biblio entries are correct.

File: index.bs [hunk 1]
```diff
@@ -490,6 +490,20 @@ For better accessibility, a web-based presentation application provides
 automatic image captioning by running a machine learning model such as
 [[im2txt]] which predicts explanatory words of the presentation slides.
 
+### Text-to-image ### {#usecase-text-to-image}
+
+Images are a core part of modern web experiences. An ability to in a
```
Comment (inexorabletash) [inline index.bs:495]: This sentence is awkward to read ("split infinitive"). Suggested rewording:

> An ability to generate images based on text input in a privacy-preserving manner enables...

or:

> An ability to generate images in a privacy-preserving manner based on text input enables...

File: index.bs [hunk 2]
```diff
@@ -520,6 +534,29 @@ noise suppression using Recurrent Neural Network such as [[RNNoise]] for
...
 
+### Speech Recognition ### {#usecase-speech-recognition}
+
+Speech recognition, also known as speech to text, enables recognition and
+translation of spoken language into text. Example applications of speech
+recognition include transcription, automatic translation, multimodal interaction,
+real-time captioning and virtual assistants. Speech recognition improves
+accessibility of auditory content and makes it possible to interact with such
+content in a privacy-preserving manner in a textual form. Examples of common
+use cases include watching videos or participating online meetings using
```
Comment (inexorabletash) [inline index.bs:545]: nit: _"...participating in online meetings..."_

File: index.bs (hunk 1 above)
Comment (anssiko) [reply index.bs]: Thanks! This _felt_ somehow wrong when I wrote it, but just couldn't figure out why or how to improve :)

Fixed by https://github.com/webmachinelearning/webnn/pull/507/commits/1d0f5c04967fdd4cec6c3da7f942d8eeb1265e1f

File: index.bs (hunk 2 above)
Comment (anssiko) [reply index.bs]: Fixed by https://github.com/webmachinelearning/webnn/pull/507/commits/1d0f5c04967fdd4cec6c3da7f942d8eeb1265e1f

**Comment by bhack** [comment]
What about Video transformers?
https://arxiv.org/abs/2201.05991

**Review by zolkis — COMMENTED** [review]
Refs look good.

**Comment by anssiko** [comment]
>What about Video transformers?
>https://arxiv.org/abs/2201.05991

@bhack, it would be helpful if you could open a new issue for the proposal similarly to #524.

**Pull request merged**